### ISSUE TITLE
Thread-local storage is not supported for the current target + AsyncDisplayKit Framework name

### ISSUE THREAD

**Issue by Kaspik** [TextureGroup/Texture#969]
Hey Texture team! I haven't found any issue here (except https://github.com/TextureGroup/Texture/issues/873).

I have problem with tests target. My main target runs okay on both XCode 9.4 and XCode 10.
But tests target on main target is not running because of:
1) Framework not found
```
Ld /Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/ClassDojo.app/ClassDojo normal x86_64 (in target: ClassDojo)
    cd /Users/kaspik/Documents/Vyvoj/STRV/dojo-behavior
    export IPHONEOS_DEPLOYMENT_TARGET=9.0
    /Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -arch x86_64 -isysroot /Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator12.0.sdk -L/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator -LVendor -LVendor/DraggableCollectionView -LVendor/MFSideMenu -LVendor/ssdatakit -LVendor/DraggableCollectionView/Helpers -LVendor/DraggableCollectionView/Layout -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator -F/Users/kaspik/Documents/Vyvoj/STRV/dojo-behavior/Vendor -F/Users/kaspik/Documents/Vyvoj/STRV/dojo-behavior -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/AFNetworking -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/AMPopTip -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/Bolts -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/CDDesignSystem -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/CocoaAsyncSocket -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/CocoaLumberjack -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/DeviceUtil -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/EFCountingLabel -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/EZAudio -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/FBSDKCoreKit -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/FLAnimatedImage -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/FirebaseCore -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/FirebaseMessaging -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/GoogleToolboxForMac -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/JDStatusBarNotification -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/JGProgressHUD -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/JSQMessagesViewController -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/JSQSystemSoundPlayer -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/KeychainAccess -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/Kingfisher -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/Layout -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/LogglyLogger-CocoaLumberjack -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/PINCache -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/PINOperation -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/PINRemoteImage-Core-PINCache-iOS -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/PINRemoteImage.default-Core -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/PNChart -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/PaperTrailLumberjack -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/PersistentStreamPlayer -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/PromiseKit -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/Protobuf -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/PubNub -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/SAMKeychain -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/SnapKit -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/SwiftyStoreKit -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/TPCircularBuffer -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/TTTAttributedLabel -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/Texture -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/Tweaks -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/UICollectionViewLeftAlignedLayout -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/UICountingLabel -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/UIFloatLabelTextField -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/UIImageViewModeScaleAspect -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/YYText -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/iRate -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/nanopb -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/pop -F/Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/zipzap-iOS -F/Users/kaspik/Documents/Vyvoj/STRV/dojo-behavior/Pods/Crashlytics/iOS -F/Users/kaspik/Documents/Vyvoj/STRV/dojo-behavior/Pods/Fabric/iOS -F/Users/kaspik/Documents/Vyvoj/STRV/dojo-behavior/Pods/FirebaseAnalytics/Frameworks -F/Users/kaspik/Documents/Vyvoj/STRV/dojo-behavior/Pods/FirebaseInstanceID/Frameworks -F/Users/kaspik/Documents/Vyvoj/STRV/dojo-behavior/Vendor -F/Users/kaspik/Documents/Vyvoj/STRV/dojo-behavior -filelist /Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Intermediates.noindex/ClassDojo.build/Testing-iphonesimulator/ClassDojo.build/Objects-normal/x86_64/ClassDojo.LinkFileList -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -Xlinker -rpath -Xlinker @executable_path/Frameworks -mios-simulator-version-min=9.0 -dead_strip -Xlinker -object_path_lto -Xlinker /Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Intermediates.noindex/ClassDojo.build/Testing-iphonesimulator/ClassDojo.build/Objects-normal/x86_64/ClassDojo_lto.o -Xlinker -export_dynamic -Xlinker -no_deduplicate -Xlinker -objc_abi_version -Xlinker 2 -fobjc-arc -fobjc-link-runtime -L/Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphonesimulator -Xlinker -add_ast_path -Xlinker /Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Intermediates.noindex/ClassDojo.build/Testing-iphonesimulator/ClassDojo.build/Objects-normal/x86_64/ClassDojo.swiftmodule -ObjC -lc++ -lsqlite3 -lz -framework AFNetworking -framework AMPopTip -framework AsyncDisplayKit -framework Bolts -framework CDDesignSystem -framework CocoaAsyncSocket -framework CocoaLumberjack -framework Crashlytics -framework DeviceUtil -framework EFCountingLabel -framework EZAudio -framework FBSDKCoreKit -framework FLAnimatedImage -framework Fabric -framework FirebaseAnalytics -framework FirebaseCore -framework FirebaseCoreDiagnostics -framework FirebaseInstanceID -framework FirebaseMessaging -framework FirebaseNanoPB -framework Foundation -framework GoogleToolboxForMac -framework JDStatusBarNotification -framework JGProgressHUD -framework JSQMessagesViewController -framework JSQSystemSoundPlayer -framework KeychainAccess -framework Kingfisher -framework Layout -framework LogglyLogger_CocoaLumberjack -framework PINCache -framework PINOperation -framework PINRemoteImage -framework PNChart -framework PaperTrailLumberjack -framework PersistentStreamPlayer -framework PromiseKit -framework Protobuf -framework PubNub -framework SAMKeychain -framework Security -framework SnapKit -framework StoreKit -framework SwiftyStoreKit -framework SystemConfiguration -framework TPCircularBuffer -framework TTTAttributedLabel -framework Tweaks -framework UICollectionViewLeftAlignedLayout -framework UICountingLabel -framework UIFloatLabelTextField -framework UIImageViewModeScaleAspect -framework UIKit -framework YYText -framework iRate -framework nanopb -framework pop -framework zipzap -Xlinker -sectcreate -Xlinker __TEXT -Xlinker __entitlements -Xlinker /Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Intermediates.noindex/ClassDojo.build/Testing-iphonesimulator/ClassDojo.build/ClassDojo.app-Simulated.xcent -framework AVFoundation -framework AdSupport -weak_framework UserNotifications -framework Photos -framework Crashlytics -framework Fabric -lsqlite3 -framework AddressBook -framework Social -framework Accounts -framework SystemConfiguration -framework Security -framework MobileCoreServices -framework CoreLocation -framework CFNetwork -framework AudioToolbox -framework UIKit -framework StoreKit -framework CoreFoundation -framework QuartzCore -lz -framework CoreGraphics -framework CoreData -framework Foundation -framework Pods_ClassDojo -Xlinker -dependency_info -Xlinker /Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Intermediates.noindex/ClassDojo.build/Testing-iphonesimulator/ClassDojo.build/Objects-normal/x86_64/ClassDojo_dependency_info.dat -o /Users/kaspik/Library/Developer/Xcode/DerivedData/ClassDojo-atbalbdthvzlqngvkcwmnjpzninu/Build/Products/Testing-iphonesimulator/ClassDojo.app/ClassDojo

ld: framework not found AsyncDisplayKit
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
Mainly issue - `ld: framework not found AsyncDisplayKit`. I'm not sure what's wrong / what's happening here as my Podfile setup looks okay and works last years without any change (Podfile included below).

2) Thread-local storage in Texture when running tests on multiple simulators
`Thread-local storage is not supported for the current target` - for example on `ASLayoutElement.mm` - `static _Thread_local __unsafe_unretained ASLayoutElementContext *tls_context;`

Is this because of iOS 9 support? I get it that it's probably related to 32bit simulators, but is it somehow possible to disable them? (only active arch is set to true - more info below). I get this error even when building tests for iPhone X.

Simplified podfile:
```
target 'ClassDojo' do
    
    pod 'Texture', '~> 2.7'
    
    target 'ClassDojo Unit Tests' do
        inherit! :complete
        pod 'OHHTTPStubs/Swift'
        pod 'FBSnapshotTestCase'
    end
    
    target 'ClassDojo UI Tests' do
        inherit! :complete
    end
end

post_install do |installer_representation|
    installer_representation.pods_project.targets.each do |target|
        
        if target.name == "Tweaks"
            target.build_configurations.each do |config|
                config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= ['$(inherited)', 'FB_TWEAK_ENABLED=1']
            end
        end
        
        if target.name == "Texture" then
            target.build_configurations.each do |configuration|
                if configuration.name.include?("Debug") then
                    configuration.build_settings['ONLY_ACTIVE_ARCH'] = 'YES'
                end
            end
        end
    end
end
```
Actually the trick with `ONLY_ACTIVE_ARCH` if from https://github.com/TextureGroup/Texture/issues/873 but still it didn't help.

3) My last question is - are you going to rename the framework on CocoaPods to Texture? Right now it's confusing that the pod is Texture but framework in Pods is AsyncDisplaKit and is still AsyncDisplayKit as module in build settings.

I'm aware that XCode 10 is early beta but still I wanted to submit this so you all are aware of that if it's an issue.
Thank you for all your help if you have any ideas! :)

**Comment by dox-mark** [comment]
I'm getting issue 2 `Thread-local storage is not supported for the current target` if I build with Texture 2.7 on my tests target as well.

**Comment by dox-mark** [comment]
Adding `configuration.build_settings['ONLY_ACTIVE_ARCH'] = 'YES'` to the pod file did fix my error with thread local storage.

**Comment by YanisSOTO** [comment]
Any update about the issue ? Dox-mark's solution doesn't work for me.

**Comment by PatrickSCLin** [comment]
Dox-mark's solution works for me~

**Comment by Kaspik** [comment]
That's great, but answer or at least any activity from Texture guys would be more appreciated.

What about you guys, @maicki, @appleguy any ideas or plans/roadmaps?

**Comment by maicki** [comment]
@Adlai-Holler Any idea how we could resolve this?

**Comment by Kaspik** [comment]
@Adlai-Holler Thanks for taking care of the 23-bit simulator issue!
Can we also help somehow with 3) that is related to 1) - AsyncDisplayKit vs Texture naming in core of the framework + CocoaPods?

**Comment by maicki** [comment]
@Kaspik You mind filing a new issue for that. This would have some wider consequences like every code need to change due to that and therefore I think needs some more discussion.

I will close this issue for now as it was mainly related to the thread local issue.

**Comment by Kaspik** [comment]
@maicki Yup, done - https://github.com/TextureGroup/Texture/issues/1045

**Comment by irodrigo17** [comment]
Hey guys, any updates on this? Every time I run `pod install` the build breaks for me because of this error. 
I think it's also related to the fact that CocoaPods is not using the recommended Xcode project settings when creating the Pods project as described here: https://github.com/CocoaPods/CocoaPods/issues/7206
Any updates would be very much appreciated, the workaround didn't work for me.

**Comment by mjjimenez** [comment]
@irodrigo17 it's fixed in #1025 not #1045. There's just no release yet. Point your pod installation to `master` branch. Don't do the work around using `Build Active Architecture`. That only works on debug build and not release.

**Comment by irodrigo17** [comment]
Oh, cool! I missed the other PR, thanks @mjjimenez! 👍

**Comment by diederikh** [comment]
Any update on this? When will #1025 be included in a release?

**Comment by Stree7** [comment]
I correct this issue by the following steps: Remove all pod related files except podfile, then restart 'pod install' , and run again.

**Comment by Adlai-Holler** [comment]
Fix released in [2.8](https://github.com/TextureGroup/Texture/releases/tag/2.8)
